src/preferences/dialog/dlgprefmixer.cpp

```diff
@@ -48,167 +70,178 @@ DlgPrefMixer::DlgPrefMixer(
-    // TODO(ronso0) Write only in slotApply(), read from config only in slotUpdate().
-    // Currently config is read in both slotUpdate() and loadSettings().
-    CheckBoxSingleEqEffect->setChecked(
-            m_pConfig->getValue(ConfigKey(kConfigGroup, kSingleEq), "yes") == "yes");
-    slotSingleEqCheckboxChanged(CheckBoxSingleEqEffect->isChecked());
-
-    // Add drop down lists for current decks and connect num_decks control
-    // to slotNumDecksChanged
-    m_pNumDecks = new ControlProxy(QStringLiteral("[App]"), QStringLiteral("num_decks"), this);
-    m_pNumDecks->connectValueChanged(this, &DlgPrefMixer::slotNumDecksChanged);
-    slotNumDecksChanged(m_pNumDecks->get());
+            &DlgPrefMixer::slotSingleEqToggled);
 
+    // Update the QuickEffect selectors when the effect list was changed
+    // in Effects preferences
     connect(m_pChainPresetManager.data(),
             &EffectChainPresetManager::quickEffectChainPresetListUpdated,
             this,
-            &DlgPrefMixer::slotPopulateDeckEffectSelectors);
+            &DlgPrefMixer::slotPopulateQuickEffectSelectors);
 
     setUpMainEQ();
 
-    setScrollSafeGuardForAllInputWidgets(this);
-
+    // Update only after all settings are loaded, except EQs and QuickEffecs.
+    // slotNumDecksChanged() needs the correct state of 'Same EQ for all decks".
     slotUpdate();
+
+    // Add drop down lists for current decks and connect to num_decks control
+    // so new lists are added if new decks are added.
+    m_pNumDecks = new ControlProxy(QStringLiteral("[App]"), QStringLiteral("num_decks"), this);
+    m_pNumDecks->connectValueChanged(this, &DlgPrefMixer::slotNumDecksChanged);
+    slotNumDecksChanged(m_pNumDecks->get());
+
+    setScrollSafeGuard(SliderXFader);
+    setScrollSafeGuard(SliderHiEQ);
+    setScrollSafeGuard(SliderLoEQ);
+    setScrollSafeGuard(comboBoxMainEq);
+
+    // This applies the Main EQ and saves default values of previously missing
+    // ConfigKeys. EQ/QuickEffects are already applied by slotNumDecksChanged().
     slotApply();
+
+    m_initializing = false;
 }
 
 DlgPrefMixer::~DlgPrefMixer() {
-    delete m_pxfScene;
-
     qDeleteAll(m_deckEqEffectSelectors);
     m_deckEqEffectSelectors.clear();
 
     qDeleteAll(m_deckQuickEffectSelectors);
     m_deckQuickEffectSelectors.clear();
 }
 
+// Create EQ & QuickEffect selectors and deck label for each added deck
 void DlgPrefMixer::slotNumDecksChanged(double numDecks) {
-    int oldDecks = m_deckEqEffectSelectors.size();
     while (m_deckEqEffectSelectors.size() < static_cast<int>(numDecks)) {
         int deckNo = m_deckEqEffectSelectors.size() + 1;
+        QString deckGroup = PlayerManager::groupForDeck(deckNo - 1);
 
         QLabel* label = new QLabel(QObject::tr("Deck %1").arg(deckNo), this);
 
-        // Create the drop down list for deck EQs
+        // Create the EQ selector //////////////////////////////////////////////
         QComboBox* pEqComboBox = new QComboBox(this);
```

Comment:
The lambda is capturing `pQuickEffectComboBox` by value (which is fine for a raw pointer, but not fine for a smart-ish-pointer such as `parented_ptr`). Thus the compiler wants to create a copy which it can't because the copy-ctor is deleted (intentional and correct for now). 
This is solved most easily by capturing by reference: `[this, &pQuickEffectComboBox]`

@daschauer do we want to rethink the `parented_ptr` semantics? Eg enabling moves and and copies since the `parented_ptr` is not really owning anyways.